dist/clouduploader.rb

```diff
@@ -76,9 +76,11 @@ def upload_image_to_ec2(image, credentials, region, filename)
   region = data['region']
   arn = data['arn']
   external_id = data['external_id']
+  ami_name = data['ami_name']
+  virtualization_type = data['virtualization_type']
 
   credentials = get_ec2_credentials(arn, external_id)
-  upload_image_to_ec2(image_path, credentials, region, filename)
+  upload_image_to_ec2(image_path, credentials, region, filename, ami_name, virtualization_type)
```

Comment:
Looks like upload_image_to_ec2 has a strong dependency on data. I would hand that variable over to upload_image_to_ec2 and extract the data there

Reply:
Makes sense, yes